x-pack/solutions/observability/plugins/slo/public/hooks/query_key_factory.ts

```diff
@@ -51,8 +51,12 @@ export const sloKeys = {
   historicalSummaries: () => [...sloKeys.all, 'historicalSummary'] as const,
   historicalSummary: (list: Array<{ sloId: string; instanceId: string }>) =>
     [...sloKeys.historicalSummaries(), list] as const,
-  definitions: (search: string, page: number, perPage: number, includeOutdatedOnly: boolean) =>
-    [...sloKeys.all, 'definitions', search, page, perPage, includeOutdatedOnly] as const,
+  definitions: (params?: {
+    search: string;
+    page: number;
+    perPage: number;
+    includeOutdatedOnly: boolean;
+  }) => [...sloKeys.all, params, 'definitions'],
```

Comment:
The order is important, 'definitions' need to come before the params object.
But I also think you will need a "allDefinitions" function that does not take any params and return `[...sloKeys.all, 'definitions']` and you should use that in the invalidateQueries of the enable/disable hook

To give more context into the cache keys, when we define the use_fetch_definitions hook, we use the keys: "slo.definitions.{params}", and on disable/enable success, we invalidate all cache entries that starts with "slo.definitions"